test/e2e/network/netpol/model.go

```diff
@@ -0,0 +1,286 @@
+	for _, namespace := range m.Namespaces {
+		for _, pod := range namespace.Pods {
+			if namespace.Name == ns && pod.Name == name {
+				return pod, nil
+			}
+		}
+	}
+	return nil, errors.Errorf("unable to find pod %s/%s", ns, name)
+}
+
+// Namespace is the abstract representation of what matters to network policy
+// tests for a namespace; i.e. it ignores kube implementation details
+type Namespace struct {
+	Name string
+	Pods []*Pod
+}
+
+// Spec builds a kubernetes namespace spec
+func (ns *Namespace) Spec() *v1.Namespace {
+	return &v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   ns.Name,
+			Labels: ns.LabelSelector(),
+		},
+	}
+}
+
+// LabelSelector returns the default labels that should be placed on a namespace
+// in order for it to be uniquely selectable by label selectors
+func (ns *Namespace) LabelSelector() map[string]string {
+	return map[string]string{"ns": ns.Name}
+}
+
+// Pod is the abstract representation of what matters to network policy tests for
+// a pod; i.e. it ignores kube implementation details
+type Pod struct {
+	Namespace  string
+	Name       string
+	Containers []*Container
+}
+
+// FindContainer returns the container matching port and protocol; otherwise, an error
+func (p *Pod) FindContainer(port int32, protocol v1.Protocol) (*Container, error) {
+	for _, cont := range p.Containers {
+		if cont.Port == port && cont.Protocol == protocol {
+			return cont, nil
+		}
+	}
+	return nil, errors.Errorf("unable to find container in pod %s/%s, port %d, protocol %s", p.Namespace, p.Name, port, protocol)
+}
+
+// PodString returns a corresponding pod string
+func (p *Pod) PodString() PodString {
+	return NewPodString(p.Namespace, p.Name)
+}
+
+// ContainerSpecs builds kubernetes container specs for the pod
+func (p *Pod) ContainerSpecs() []v1.Container {
+	var containers []v1.Container
+	for _, cont := range p.Containers {
+		containers = append(containers, cont.Spec())
+	}
+	return containers
+}
+
+func (p *Pod) labelSelectorKey() string {
+	return "pod"
+}
+
+func (p *Pod) labelSelectorValue() string {
+	return p.Name
+}
+
+// LabelSelector returns the default labels that should be placed on a pod/deployment
+// in order for it to be uniquely selectable by label selectors
+func (p *Pod) LabelSelector() map[string]string {
+	return map[string]string{
+		p.labelSelectorKey(): p.labelSelectorValue(),
+	}
+}
+
+// KubePod returns the kube pod
+func (p *Pod) KubePod() *v1.Pod {
+	zero := int64(0)
+	return &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      p.Name,
+			Labels:    p.LabelSelector(),
+			Namespace: p.Namespace,
+		},
+		Spec: v1.PodSpec{
+			TerminationGracePeriodSeconds: &zero,
+			Containers:                    p.ContainerSpecs(),
+		},
+	}
+}
+
+// QualifiedServiceAddress returns the address that can be used to hit a service from
+// any namespace in the cluster
+func (p *Pod) QualifiedServiceAddress() string {
+	return fmt.Sprintf("%s.%s.svc.cluster.local", p.ServiceName(), p.Namespace)
```

Comment:
there was a bug fixed recently because we were assuming that the cluster always have the domain cluster.local, but that is not true. See https://github.com/kubernetes/kubernetes/pull/92964

Reply:
ahhh your right, I see `framework.TestContext.ClusterDNSDomain` seems to be the idiomatic way to do this in the e2e's.

Reply:
ok I just fixed this, it now is 

```
test/e2e/network/netpol/model.go:func (p *Pod) QualifiedServiceAddress(dnsDomain string) string {
```

and grabs it from the framework meta